src/main/java/seedu/duke/commands/Add.java

```diff
@@ -0,0 +1,125 @@
+import seedu.duke.exceptions.InvalidInputFormatException;
+
+public class Add extends Command {
+    private Module mod;
+
+    /**
+     * Contructor of Add class to initialize an object of class Add.
+     * @param input the input message to be used to initialize the variables
+     * @throws InvalidInputFormatException exception which is thrown if the format of the input is wrong
+     * @throws InvalidInputContentException exception to be thrown if the input content is empty
+     */
+    public Add(String input) throws InvalidInputFormatException, InvalidInputContentException {
+        checkFormat(input);
+        int[] indexes = positions(input);
+        checkContent(input, indexes);
+        addition(input, indexes);
+    }
+
+    /**
+     * Function to extract details from input text and create a new module with the details extracted.
+     * Extracted details include course, semester, mc, and grade.
+     * @param input the input entered by the user
+     * @param indexes an array containing the positions from which the details need to be extracted
+     */
+    private void addition(String input, int[] indexes) {
+        String course = extractingContent(input, indexes[0], indexes[1]);
+        String semester = extractingContent(input, indexes[2], indexes[3]);
+        String mcString = extractingContent(input, indexes[4], indexes[5]);
+        int mc = Integer.parseInt(mcString);
+        String grade = extractingContent(input, indexes[6], indexes[7]);
+
+        this.mod = new Module(course.toUpperCase(), semester.toUpperCase(), grade.toUpperCase(), mc);
+    }
+
+    /**
+     * Function to extract content from startIndex to endIndex out of input
+     * @param input The input from which the content needs to be extracted. Format: String
+     * @param startIndex The starting index. Format: int
+     * @param endIndex The ending index. Format: int
+     * @return a string which is a substring (extracted) of input.
+     */
+    private String extractingContent(String input, int startIndex, int endIndex) {
+        if (endIndex == -1) {
+            return input.substring(startIndex);
+        } else {
+            return input.substring(startIndex, endIndex);
+        }
+    }
+
+    /**
+     * Function to check if the format of input is correct or not
+     * @param input input entered by user. Format: String
+     * @throws InvalidInputFormatException exception thrown if format of input is incorrect
+     */
+
+    public void checkFormat(String input) throws InvalidInputFormatException {
+        boolean isRight;
+        isRight = InvalidInputFormatException.checkMod(input);
+        checkFormatException(isRight);
+        isRight = InvalidInputFormatException.checkSem(input);
+        checkFormatException(isRight);
+        isRight = InvalidInputFormatException.checkMC(input);
+        checkFormatException(isRight);
+        isRight = InvalidInputFormatException.checkGrade(input);
+        checkFormatException(isRight);
```

Comment:
Avoid reusing variables. In this case, if-elif-else statements could be used to better structure the logic, which would remove the boolean variable altogether. This allows you to state different types of invalid input as well! For example: 
```
if (!InvalidInputFormatException.checkMod(input)) { 
    throw new InvalidInputFormatException()
} else if (...) {...}
```

This applies to the similiarly structure input checks.